zebra-consensus/src/chain.rs

```diff
@@ -103,27 +107,27 @@ where
         Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send + 'static>>;
 
     fn poll_ready(&mut self, _: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
-        // We don't expect the state or verifiers to exert backpressure on our
-        // users, so we don't need to call `state_service.poll_ready()` here.
+        // We don't expect the verifiers to exert backpressure on our
+        // users, so we don't need to call the verifier's `poll_ready` here.
         // (And we don't know which verifier to choose at this point, anyway.)
         Poll::Ready(Ok(()))
     }
 
     fn call(&mut self, block: Arc<Block>) -> Self::Future {
-        // TODO(jlusby): Error = Report, handle errors from state_service.
+        // TODO(jlusby): Error = Report
```

Comment:
Should we remove this TODO? It seemed like the conclusion was that Tower's middleware is mostly standardized on `Box<dyn Error + Send + Sync + 'static>`, and I think a previous attempt to use `Report` got stuck on generic bounds.  If we don't have an immediate plan to change the error types, it might be good to remove the TODOs.  We can always work through the code and change them if the Tower situation changes.

Reply:
yea we can probably remove these TODOs. The particular issue that prevents this from being possible is that `Buffer` and `Batch` both needs error types that are `Clone`, which is just not possible to implement for Report. What we probably want to do instead is switch all of these service error types to use `TracedError` so that we can propagate spantraces back out of the service which then get used in the report.

Reply:
Got it.  For now, I think it's probably easiest to keep using `BoxError`, because we know there's no problems, even if it doesn't retain all the context we'd like, and experiment with using `TracedError` after we ship a first version of Zebra.

Reply:
I opened #1034 to make this change later.

I'm concerned that it will modify a few different files, and this PR already has conflicts with other PRs.